syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py

```diff
@@ -504,13 +504,11 @@ def register_pending(self, config: Configuration, milestone=None):
             if pos_cand is not None:
                 evals = state.candidate_evaluations
                 num_labeled = len(evals)
-                error_msg = """
-                This configuration is already registered as labeled:
-                   Position of labeled candidate: {} of {}
-                   Label value: {}
-                """.format(
-                    pos_cand, num_labeled,
-                    evals[pos_cand].metrics[INTERNAL_METRIC_NAME])
+                config_labeled = evals[pos_cand].candidate
+                error_msg = \
+                    "This configuration is already registered as labeled:\n" +\
+                    f"config:\n   {config}\n" +\
+                    f"labeled config [{pos_cand} of {num_labeled}]:\n   {config_labeled}"
                 assert False, error_msg
```

Comment:
If you really want to fail, it may be better to do:
```suggestion
                raise ValueError(error_msg)
```
since assert can be deactivated depending on how python is started.

Reply:
Done